apiserver/common/secrets/drain.go

```diff
@@ -39,24 +39,36 @@ func NewSecretsDrainAPI(
 	logger loggo.Logger,
 	leadershipChecker leadership.Checker,
 	model Model,
-	secretsState SecretsMetaState,
-	secretsConsumer SecretsConsumer,
+	secretService SecretService,
 	watcherRegistry facade.WatcherRegistry,
 ) (*SecretsDrainAPI, error) {
-	if !authorizer.AuthUnitAgent() && !authorizer.AuthApplicationAgent() && !authorizer.AuthController() {
+	if !authorizer.AuthUnitAgent() && !authorizer.AuthController() {
 		return nil, apiservererrors.ErrPerm
 	}
 	return &SecretsDrainAPI{
 		authTag:           authTag,
 		logger:            logger,
 		leadershipChecker: leadershipChecker,
 		model:             model,
-		secretsState:      secretsState,
-		secretsConsumer:   secretsConsumer,
+		secretService:     secretService,
 		watcherRegistry:   watcherRegistry,
 	}, nil
 }
 
+func filterSecret(controllerUUID, activeBackend string, md *coresecrets.SecretMetadata, rev *coresecrets.SecretRevisionMetadata) bool {
```

Comment:
Comment to explain the logic here or have a better name.

Reply:
done